### PR TITLE
Update the v0.20.0 changelog for release

### PR DESCRIPTION
**Pull request by WilliamHPNielsen**
@astafan8, @trevormorgan

### CODE REVIEW

File: docs/changes/0.20.0.rst [hunk 1]
```diff
@@ -1,43 +1,76 @@
...
+- The ``Measurement`` now captures exception info in the ``DataSet`` s metadata if an exception occured during measurement (#2327)
 
+---------
 Improved:
-_________
-
-
+---------
+- The documentation for ``ParameterWithSetpoints`` now shows how to make sweeps (#2351)
+- There is now an explicit API for marking the ``cache`` of a ``Parameter`` as invalid (#2012)
```
Comment (astafan8) [inline docs/changes/0.20.0.rst:23]: this is "new"

File: docs/changes/0.20.0.rst [hunk 2]
```diff
@@ -1,43 +1,76 @@
...
 Under the hood:
-_______________
-
-
-
-Dependencies:
-_____________
+---------------
+
+- The ``DataSet`` shape is used when loading data (#2264)
```
Comment (astafan8) [inline docs/changes/0.20.0.rst:48]: i'm tempted to put in in the "new" and mention that DataSet.get_parameter_data and DataSet.cache.data are the "affected" methods

File: docs/changes/0.20.0.rst [hunk 3]
```diff
@@ -1,43 +1,76 @@
...
+- Legacy datasets can have a name (#2328)
+- The deprecated HP33210A driver has been removed (#2334)
+- The deprecated Tektronix AWG5200 driver has been removed (#2335)
+- Typing was added to the drivers of the following vendorss:
```
Comment (astafan8) [inline docs/changes/0.20.0.rst:52]:
```suggestion
- Typing was added to the drivers of the following vendors:
```

File: docs/changes/0.20.0.rst (hunk 2 above)
Comment (jenshnielsen) [reply docs/changes/0.20.0.rst:48]: Agreeing with @astafan8 I would perhaps also put a link to the relevant part of the "performing measurements with ...` notebook

File: docs/changes/0.20.0.rst [hunk 4]
```diff
@@ -1,43 +1,76 @@
...
-
-
-Dependencies:
-_____________
+---------------
+
+- The ``DataSet`` shape is used when loading data (#2264)
+- Legacy datasets can have a name (#2328)
+- The deprecated HP33210A driver has been removed (#2334)
+- The deprecated Tektronix AWG5200 driver has been removed (#2335)
```
Comment (jenshnielsen) [inline docs/changes/0.20.0.rst:51]: Removal of deprecated drivers should imho be listed as breaking changes

**Review by astafan8 — APPROVED** [review]

**Pull request merged**